autotx/agents/ResearchTokensAgent.py

```diff
@@ -87,67 +93,114 @@ def _run(self, token_id: str):
             }
         )
 
+
+class SearchToken(BaseTool):
+    name: str = "search_token"
+    description: str = dedent(
+        """
+        Search token based on its symbol. It will return the ID of tokens with most market cap
+
+        Args:
+            token_symbol (str): Symbol of token to search
+            retrieve_all (bool): Only used to retrieve all tokens (Which can happen if they share the same symbol).
+                By default use False unless it's necessary to use True
+        """
+    )
+
+    def _run(self, token_symbol: str, retrieve_all: str):
```

Comment:
Shouldn't `retrieve_all` be bool?

Reply:
yes it should but it seems that the LLM always pass the argument as string even tho i explicitly mentioned it to be a boolean :disappointed: